src/meta/src/manager/hash_mapping.rs

```diff
@@ -91,6 +93,25 @@ impl HashMappingManager {
             .map(|info| info.vnode_mapping.clone())
     }
 
+    pub fn check_sst_deprecated(&self, vnode_bitmaps: &[VNodeBitmap]) -> bool {
+        let core = self.core.lock();
+        for vnode_bitmap in vnode_bitmaps {
+            if let Some(fragment_id) = core
+                .state_table_fragment_mapping
+                .get(&vnode_bitmap.get_table_id())
+            {
+                if let Some(HashMappingInfo { map_seq, .. }) =
+                    core.hash_mapping_infos.get(fragment_id)
+                {
+                    if *map_seq != vnode_bitmap.get_map_seq() {
+                        return true;
+                    }
```

Comment:
> `map_seq` will not be compared

What do you mean by "not compared"? Here you check whether `map_seq`s are equal, right? 🤔

Reply:
> > `map_seq` will not be compared
> 
> What do you mean by "not compared"? Here you check whether `map_seq`s are equal, right? 🤔

not compared by order

> if we treat a deprecated SST as not deprecated by coincidence, that deprecated SST will still be consolidated by later compactions including it